app/services/merge_requests/update_service.rb

```diff
@@ -23,11 +23,15 @@ def execute(merge_request)
         merge_request.update_nth_task(params[:task_num].to_i, false)
       end
 
+      old_labels = merge_request.labels.map{ |label| label }
```

Comment:
If I don't create an array again, `old_labels` will hold a reference to `merge_request.labels` and will be the same as the new labels after the attributes update.